src/client/client.ts

```diff
@@ -143,6 +151,121 @@ class OAuth2 implements IOAuth2 {
     };
   }
 
+  /**
+   *  {@inheritdoc}
+   */
+  async start(
+    authorizeUrlFn: OAuth2StartAuthorizeUrlFn,
+    codeAcquisitionPattern: RegExp,
+    options?: OAuth2StartOptions
+  ): Promise<OAuth2Start> {
+    const timeout = options?.timeout ?? (600 * 1000); // 10 minute timeout
+    const pollInterval = options?.pollInterval ?? 2000; // 2 second poll interval
+
+    const start = await this.client.startOAuth2Flow(
+      codeAcquisitionPattern,
+      timeout
+    );
+
+    // build our "proper" authorize URL using downstream implementation
+    const idpAuthorizeUrl = authorizeUrlFn({
+      state: start.state,
+      redirectUri: start.gatewayCallbackUrl,
+      codeChallenge: start.codeChallenge,
+    });
+
+    // formulate the authorize URL that the app user should use to that the grant flow
+    const authorizeUrl = new URL(start.gatewayAuthorizeUrl);
+    authorizeUrl.searchParams.set("return", idpAuthorizeUrl);
+    authorizeUrl.searchParams.set("callback", start.deskproCallbackUrl);
+
+    // curried poll function that returns the promise that resolves when polling is done... or rejects
+    // when polling times out
+    const poll = () => new Promise<OAuth2StartPollResult>((resolve, reject) => {
```

Comment:
Is there a way to cancel the poll that isn't just typing out. E.g. a user navigates to a different page as they want to use API key instead of Oauth2.